### PR TITLE
Improve API docs

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
$subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/2580

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

**Comment by daneshk** [comment]
Shall we change the usage of `stream.foreach` to query syntax, if possible.

File: ballerina/Module.md [hunk 1]
```diff
@@ -344,9 +346,10 @@ sql:ProcedureCallResult result =
                          check dbClient->call(`call InsertPerson(${uid}, ${insertId})`);
 stream<record{}, sql:Error?>? resultStr = result.queryResult;
 if resultStr is stream<record{}, sql:Error?> {
-    sql:Error? e = resultStr.forEach(function(record{} result) {
-      // Can perform operations using the record 'result'.
-    });
+    error? e = check from record{} value in streamData
```
Comment (daneshk) [inline ballerina/Module.md:349]:
```suggestion
    error? e = check from record{} value in resultStr
```

**Review by daneshk — DISMISSED** [review]

File: ballerina/Module.md [hunk 2]
```diff
@@ -242,14 +242,15 @@ sql:ParameterizedQuery query = `SELECT * FROM students
 stream<Student, sql:Error?> resultStream = dbClient->query(query);
 
 // Iterating the returned table.
-error? e = resultStream.forEach(function(Student student) {
-   //Can perform operations using the record 'student' of type `Student`.
-});
+error? e = check from Student student in resultStream
```
Comment (niveathika) [inline ballerina/Module.md:245]: Why is this returning an error and using check at the same time?

Comment (kaneeldias) [reply ballerina/Module.md]: Fixed in 96f6de7

**Review by niveathika — DISMISSED** [review]

**Review by daneshk — DISMISSED** [review]

**Comment by daneshk** [comment]
@praneesha Please review

File: ballerina/Module.md [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ## Overview
 
 This module provides the generic interface and functionality to interact with a SQL database. The corresponding database
```
Comment (praneesha) [inline ballerina/Module.md:3]:
```suggestion
This module provides the generic interface and functionality to interact with an SQL database. The corresponding database
```

File: ballerina/Module.md [hunk 4]
```diff
@@ -31,9 +31,9 @@ connection pool handling.  For its properties and possible values, see the [`sql
 
 2. Client-owned, unsharable connection pool
 
-    If you define the `connectionPool` field inline when creating the database client with the `sql:ConnectionPool` type, 
-    an unsharable connection pool will be created. The JDBC module sample below shows how the global 
-    connection pool is used.
+   If you define the `connectionPool` field inline when creating the database client with the `sql:ConnectionPool` type,
+   an unsharable connection pool will be created. The JDBC module sample below shows how the global
```
Comment (praneesha) [inline ballerina/Module.md:35]:
```suggestion
   an unsharable connection pool will be created. The `jdbc` module sample below shows how the global
```

File: ballerina/Module.md [hunk 5]
```diff
@@ -43,9 +43,9 @@ connection pool handling.  For its properties and possible values, see the [`sql
 
 3. Local, shareable connection pool
 
-    If you create a record of the `sql:ConnectionPool` type and reuse that in the configuration of multiple clients, 
-    for each set of clients that connects to the same database instance with the same set of properties, a shared 
-    connection pool will be created. The JDBC module sample below shows how the global connection pool is used.
+   If you create a record of the `sql:ConnectionPool` type and reuse that in the configuration of multiple clients,
+   for each set of clients that connects to the same database instance with the same set of properties, a shared
```
Comment (praneesha) [inline ballerina/Module.md:47]:
```suggestion
   a shared
```

File: ballerina/Module.md [hunk 6]
```diff
@@ -43,9 +43,9 @@ connection pool handling.  For its properties and possible values, see the [`sql
...
+   connection pool will be created. The JDBC module sample below shows how the global connection pool is used.
```
Comment (praneesha) [inline ballerina/Module.md:48]:
```suggestion
   connection pool will be created for each set of clients that connects to the same database instance with the same set of properties. The `jdbc` module sample below shows how the global connection pool is used.
```

File: ballerina/Module.md [hunk 7]
```diff
@@ -151,21 +151,21 @@ sql:ExecutionResult result =
...
 ```ballerina
 sql:ExecutionResult result = check dbClient->execute(`INSERT INTO student(age, name)
                                                         VALUES (23, 'john')`);
 ```
 
-In this sample, the parameter values, which are in local variables are used to parameterize the SQL query in 
-the `execute` remote function. This parameterization can be performed with any primitive Ballerina type 
-like `string`, `int`, `float`, or `boolean` and in that case, the corresponding SQL type of the parameter is derived 
-from the type of the Ballerina variable that is passed in. 
+In this sample, the parameter values, which are in local variables are used to parameterize the SQL query in
```
Comment (praneesha) [inline ballerina/Module.md:165]:
```suggestion
In this sample, the parameter values, which are assigned to local variables are used to parameterize the SQL query in
```

File: ballerina/Module.md [hunk 8]
```diff
@@ -220,8 +220,8 @@ Note the mapping of the database column to the returned record's property is cas
 record(i.e., the `ID` column in the result can be mapped to the `id` property in the record). Additional Column names
 added to the returned record as in the SQL query. If the record is defined as a close record, only defined fields in the
 record are returned or gives an error when additional columns present in the SQL query. Next, the `SELECT` query is executed
-via the `query` remote function of the client. Once the query is executed, each data record can be retrieved by looping 
-the result set. The `stream` returned by the select operation holds a pointer to the actual data in the database and it 
+via the `query` remote function of the client. Once the query is executed, each data record can be retrieved by looping
+the result set. The `stream` returned by the select operation holds a pointer to the actual data in the database and it
```
Comment (praneesha) [inline ballerina/Module.md:224]:
```suggestion
the result set. The `stream` returned by the `select` operation holds a pointer to the actual data in the database and it
```

File: ballerina/Module.md [hunk 9]
```diff
@@ -312,8 +314,8 @@ sql:ExecutionResult result = check dbClient->execute(query);
 
 #### Batch Updating Data
 
-This sample demonstrates how to insert multiple records with a single `INSERT` statement that is executed via the 
-`batchExecute` remote function of the client. This is done by creating a `table` with multiple records and 
+This sample demonstrates how to insert multiple records with a single `INSERT` statement that is executed via the
+`batchExecute` remote function of the client. This is done by creating a `table` with multiple records and
 parameterized SQL query as same as the above `execute` operations.
```
Comment (praneesha) [inline ballerina/Module.md:319]:
```suggestion
a parameterized SQL query as same as the above `execute` operations.
```

File: ballerina/connection-pool.bal [hunk 10]
```diff
@@ -20,17 +20,17 @@ configurable int maxOpenConnections = 15;
 configurable decimal maxConnectionLifeTime = 1800.0;
 configurable int minIdleConnections = 15;
 
-# Represents the properties which are used to configure DB connection pool.
+# Represents the properties which are used to configure a DB connection pool.
```
Comment (praneesha) [inline ballerina/connection-pool.bal:23]:
```suggestion
# Represents the properties, which are used to configure a DB connection pool.
```

File: ballerina/connection-pool.bal [hunk 11]
```diff
@@ -20,17 +20,17 @@ configurable int maxOpenConnections = 15;
...
 # Default values of the fields can be set through the configuration API.
 #
-# + maxOpenConnections - The maximum number of open connections that the pool is allowed to have including
-#                        both idle and in-use connections. The default value is 15. This can be changed through the
-#                        configuration API with the `ballerina.sql.maxOpenConnections` key
+# + maxOpenConnections - The maximum number of open connections that the pool is allowed to have.
```
Comment (praneesha) [inline ballerina/connection-pool.bal:26]:
```suggestion
# + maxOpenConnections - The maximum number of open connections that the pool is allowed to have
```
Please check the guidelines.if we need this fullstop or not.

File: ballerina/error.bal [hunk 12]
```diff
@@ -43,17 +43,17 @@ public type Error distinct error;
 # database resource clean-up, and other similar scenarios.
 public type DatabaseError distinct (Error & error<DatabaseErrorDetail>);
 
-# Represents an error occurred during execution of batch queries.
+# Represents an error that occurs during the execution of batch queries.
 public type BatchExecuteError distinct (Error & error<BatchExecuteErrorDetail>);
 
-# Represents an error that occurs when a query retrieves no rows when at most one row is expected.
+# Represents an error that occurs when a query retrieves no rows when at least one row is expected.
```
Comment (praneesha) [inline ballerina/error.bal:49]:
```suggestion
# Represents an error that occurs when a query does not retrieve any rows when at least one row is expected.
```

File: ballerina/types.bal [hunk 13]
```diff
@@ -1355,9 +1355,9 @@ public type ParameterizedCallQuery distinct object {
 
 # The result iterator used to iterate results in stream returned from `query` function.
 #
-# + customResultIterator - Any custom result iterator to be used overriding the default behaviour
-# + err - Used to hold any error occurring at the instance of stream creation
-# + isClosed - Indicated the stream state
+# + customResultIterator - Any custom result iterator to be used to override the default behaviour
+# + err - Used to hold any error that occurs at the instance of stream creation
```
Comment (praneesha) [inline ballerina/types.bal:1359]:
```suggestion
# + err - Used to hold any error that occurs at the instance of the stream creation
```

File: ballerina/types.bal [hunk 14]
```diff
@@ -1410,11 +1410,11 @@ public class ResultIterator {
     }
 }
 
-# Represents the results from `call` method holding returned results or metadata of query execution.
+# Represents the results from `call` method holding returned results or metadata of the query execution.
```
Comment (praneesha) [inline ballerina/types.bal:1413]:
```suggestion
# Represents the results from the `call` method holding the returned results or metadata of the query execution.
```

File: ballerina/types.bal [hunk 15]
```diff
@@ -1435,15 +1435,15 @@ public class ProcedureCallResult {
...
+    # Releases the associated resources such as the database connection, results, etc.
     #
     # + return - An `sql:Error` if any error occurred while cleanup
     public isolated function close() returns Error? {
         return closeCallResult(self);
     }
 }
 
-# The iterator for the stream returned in `query` function to be used overriding the default behaviour of sql:ResultIterator.
+# The iterator for the stream returned in `query` function to be used to override the default behaviour of `sql:ResultIterator`.
```
Comment (praneesha) [inline ballerina/types.bal:1446]:
```suggestion
# The iterator for the stream returned in the `query` function to be used to override the default behavior of the `sql:ResultIterator`.
```

File: ballerina/connection-pool.bal (hunk 11 above)
Comment (kaneeldias) [reply ballerina/connection-pool.bal:26]: We need a fullstop as the description for the parameter is multiple sentences.

Comment (praneesha) [reply ballerina/connection-pool.bal:26]: Isn't this the only sentence we have in the description? `The maximum number of open connections that the pool is allowed to have.`

File: ballerina/types.bal (hunk 15 above)
Comment (kaneeldias) [reply ballerina/types.bal]: Fixed in 3c1e3f7

File: ballerina/types.bal (hunk 14 above)
Comment (kaneeldias) [reply ballerina/types.bal]: Fixed in 3c1e3f7

File: ballerina/types.bal (hunk 13 above)
Comment (kaneeldias) [reply ballerina/types.bal]: Fixed in 3c1e3f7

File: ballerina/error.bal (hunk 12 above)
Comment (kaneeldias) [reply ballerina/error.bal]: Fixed in 3c1e3f7

File: ballerina/Module.md (hunk 8 above)
Comment (kaneeldias) [reply ballerina/Module.md]: Fixed in 3c1e3f7

File: ballerina/Module.md (hunk 9 above)
Comment (kaneeldias) [reply ballerina/Module.md:323]: Fixed in 3c1e3f7

File: ballerina/connection-pool.bal (hunk 10 above)
Comment (kaneeldias) [reply ballerina/connection-pool.bal]: Fixed in 3c1e3f7

File: ballerina/Module.md (hunk 7 above)
Comment (kaneeldias) [reply ballerina/Module.md]: Fixed in 3c1e3f7

File: ballerina/Module.md (hunk 6 above)
Comment (kaneeldias) [reply ballerina/Module.md]: Fixed in 3c1e3f7

File: ballerina/Module.md (hunk 5 above)
Comment (kaneeldias) [reply ballerina/Module.md]: Fixed in 3c1e3f7

File: ballerina/Module.md (hunk 3 above)
Comment (kaneeldias) [reply ballerina/Module.md]: Fixed in 3c1e3f7

**Review by praneesha — APPROVED** [review]

**Pull request merged**